Library/TeamTalkLib/teamtalk/client/ClientNode.cpp

```diff
@@ -3010,14 +3010,30 @@ bool ClientNode::EnableAudioBlockCallback(int userid, StreamTypes sts,
     {
         if (enable)
         {
-            m_audiomuxer_stream.reset(new AudioMuxer(sts));
-            // start muxer if already in channel
-            if (m_mychannel)
+            audiomuxer_t newmuxer(new AudioMuxer(sts));
+            media::AudioInputFormat infmt;
+
+            if (outfmt.IsValid())
+                infmt = media::AudioInputFormat(outfmt, PCM16_DURATION_SAMPLES(20, outfmt.samplerate));
+            else if (m_mychannel)
             {
                 auto codec = m_mychannel->GetAudioCodec();
-                bool startmux = m_audiomuxer_stream->RegisterMuxCallback(codec, std::bind(&ClientNode::AudioMuxCallback, this, _1, _2));
+                infmt = GetAudioCodecAudioInputFormat(codec);
+            }
```

Comment:
If not output format is specified, i.e. using `TT_EnableAudioBlockEvent` instead of `TT_EnableAudioBlockEventEx` while not being in a channel will return "false" with this modification. Should we just default to 48000Hz/Stereo?

Reply:
I'd be OK with 48K/stereo, as long as it's a documented behavior.